spring-cloud-netflix-core/src/test/java/org/springframework/cloud/netflix/hystrix/HystrixCircuitBreakerTest.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2013-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.netflix.hystrix;
+
+import org.junit.Test;
+import org.springframework.cloud.client.circuitbreaker.CircuitBreaker;
+
+import static org.junit.Assert.assertEquals;
+
+
+/**
+ * @author Ryan Baxter
+ */
+public class HystrixCircuitBreakerTest {
+
+	@Test
+	public void run() {
+		HystrixCircuitBreakerConfigFactory configFactory = new HystrixCircuitBreakerConfigFactory.DefaultHystrixCircuitBreakerConfigFactory();
```

Comment:
Why isn't this just part of the builder? The default could be there if not set. I'd also add `HystrixCommand.Setter` as part of the builder similar to what we did with [HystrixCommands](https://github.com/spring-cloud/spring-cloud-netflix/blob/master/spring-cloud-netflix-hystrix/src/main/java/org/springframework/cloud/netflix/hystrix/HystrixCommands.java).

Reply:
You mean why is it passed in the constructor as opposed to as a method in the builder?  It is passed in the builder constructor now.  

I suppose we could experiment with adding a method to the builder interface and using type params to pass it as a parameter in the method.  The config object is going to be different for different implementations.  I am not sure what the benefit is of that though.

Reply:
If a builder doesn't have more methods than that in the interface, not sure how useful it is.